tests/testthat/test-alphaDiv.R

```diff
@@ -0,0 +1,133 @@
+  attr <- attributes(results)
+  expect_true(all(c('computationDetails','parameters','recordVariable','computedVariableDetails') %in% names(attr)))
+  expect_equal(names(attr$computedVariableDetails), c('variableId','entityId','dataType','dataShape','displayLabel','defaultRange'))
+  expect_equal(attr$computedVariableDetails$variableId, 'alphaDiversity')
+  expect_equal(attr$computedVariableDetails$entityId, 'entity')
+  expect_equal(attr$computedVariableDetails$displayLabel, 'Shannon')
+  
+  results <- alphaDiv(df, "entity.SampleID", method='simpson', verbose=F)
+  attr <- attributes(results)
+  expect_true(all(c('computationDetails','parameters','recordVariable','computedVariableDetails') %in% names(attr)))
+  expect_equal(names(attr$computedVariableDetails), c('variableId','entityId','dataType','dataShape','displayLabel','defaultRange'))
+  expect_equal(attr$computedVariableDetails$variableId, 'alphaDiversity')
+  expect_equal(attr$computedVariableDetails$entityId, 'entity')
+  expect_equal(attr$computedVariableDetails$displayLabel, 'Simpson')
+  
+  results <- alphaDiv(df, "entity.SampleID", method='evenness', verbose=F)
+  attr <- attributes(results)
+  expect_true(all(c('computationDetails','parameters','recordVariable','computedVariableDetails') %in% names(attr)))
+  expect_equal(names(attr$computedVariableDetails), c('variableId','entityId','dataType','dataShape','displayLabel','defaultRange'))
+  expect_equal(attr$computedVariableDetails$variableId, 'alphaDiversity')
+  expect_equal(attr$computedVariableDetails$entityId, 'entity')
+  expect_equal(attr$computedVariableDetails$displayLabel, 'Pielou\'s Evenness')
+  
+})
+
+test_that("alphaDivApp produces an appropriately structured list of computations", {
+  
+  df <- testOTU
+  
+  # Default - use all methods
+  appResults <- alphaDivApp(df, "entity.SampleID", verbose=F)
+  expect_equal(length(appResults), 3)
+  expect_equal(unique(unlist(lapply(appResults,names))), c('entity.SampleID','entity.alphaDiversity'))
+  expect_equal(unique(unlist(lapply(appResults, class))), c('data.table','data.frame'))
+  expect_equal(unique(unlist(lapply(appResults, NROW))), NROW(df))
+  
+  # Test using a subset of methods
+  appResults <- alphaDivApp(df, "entity.SampleID", methods=c('evenness','simpson'), verbose=F)
+  expect_equal(length(appResults), 2)
+  expect_equal(unique(unlist(lapply(appResults,names))), c('entity.SampleID','entity.alphaDiversity'))
+  expect_equal(unique(unlist(lapply(appResults, class))), c('data.table','data.frame'))
+  expect_equal(unique(unlist(lapply(appResults, NROW))), NROW(df))
+  
+})
+
+test_that("alphaDivApp output is correctly represented in json", {
+  
+  df <- testOTU
+  
+  # Default - use all methods
+  appResults <- alphaDivApp(df, "entity.SampleID", verbose=F)
+  outJson <- getAppJson(appResults)
+  jsonList <- jsonlite::fromJSON(outJson)
+  expect_equal(names(jsonList), c('computations','parameterSets'))
```

Comment:
this seems sparse. things like parameterSets are basically just said to exist, but their structure untested

Reply:
recordVariableDetails another ex

Reply:
hmm this brings up another question - would the column for sampleId/recordId ever be different across computations? I don't think it could be. The main app function definitely isn't set up to have multiple recordIds. So note to me i need to move recordVariableDetails to a higher level